binsync/core/git_actions_util.py

```diff
@@ -0,0 +1,38 @@
+from functools import wraps
+
+def atomic_git_action(f):
```

Comment:
I don't 100% know what your plan is yet for this decorator, but if this decorator is ever only used by one object (like GitServer or something), then keep it in the same file as that object definition

Reply:
Okay so question, will the client need atomic git actions once the server implementation exists, since object mode is basically just a client. My thought process was that logically the clients would just forward their git states over to the server through RPC, and the server would take those states and perform the same git actions using the decorator. I guess my question is which one needs to handle atomic git actions? I don't think its both? 

You're absolutely right though it should be kept in the file its being used in, I just took it out so I could figure out which one it needed to be in